src/NuGet.Indexing/Extraction/CatalogPackageMetadataExtraction.cs

```diff
@@ -64,12 +83,31 @@ private class Extractor
                 AddString(MetadataConstants.CatalogMetadata.RequiresLicenseAcceptancePropertyName, MetadataConstants.RequiresLicenseAcceptancePropertyName);
 
                 AddLicenseUrl();
+                AddIconUrl(flatContainerBaseAddress, flatContainerContainerName);
                 AddFlattenedDependencies();
                 AddSupportedFrameworks();
 
                 return _metadata;
             }
 
+            private void AddIconUrl(Uri flatContainerBaseAddress, string flatContainerContainerName)
+            {
+                var iconFile = JTokenToString(_catalog[MetadataConstants.IconFilePropertyName]);
+                if (!string.IsNullOrWhiteSpace(iconFile))
+                {
+                    var packageId = JTokenToString(_catalog[MetadataConstants.IdPropertyName]);
+                    var packageVersion = JTokenToString(_catalog[MetadataConstants.NormalizedVersionPropertyName]);
+
+                    var pathProvider = new FlatContainerPackagePathProvider(flatContainerContainerName);
```

Comment:
>FlatContainerPackagePathProvider [](start = 43, length = 32)

Can't we just pass the provider down instead of two params?

Reply:
The choice was between:
* `Uri flatContainerBaseAddress, string flatContainerContainerName`;
* `Uri flatContainerBaseAddress, IPackagePathProvider pathProvider`;
I.e. we can't get rid of `flatContainerBaseAddress` (see the few lines below, path provider does not provide full URL). Passing `IPackagePathProvider` provides a bit more freedom than I want in this method: it only supports flat container paths, so path provider must be `FlatContainerPackagePathProvider`.

Reply:
Sounds good!